src/utils/validateBurnedAmount.ts

```diff
@@ -1,136 +1,83 @@
 import type { Context } from '../vms/context/model';
-import {
-  isAddDelegatorTx,
-  isAddPermissionlessDelegatorTx,
-  isAddPermissionlessValidatorTx,
-  isAddSubnetValidatorTx,
-  isAddValidatorTx,
-  isCreateChainTx,
-  isCreateSubnetTx,
-  isPvmBaseTx,
-  isExportTx as isPvmExportTx,
-  isImportTx as isPvmImportTx,
-  isRemoveSubnetValidatorTx,
-  isTransferSubnetOwnershipTx,
-  isTransformSubnetTx,
-} from '../serializable/pvm';
 import type { Transaction, UnsignedTx } from '../vms/common';
 import type { EVMTx } from '../serializable/evm';
 import { isImportExportTx as isEvmImportExportTx } from '../serializable/evm';
-import { costCorethTx } from './costs';
+import { getBurnedAmountByTx } from './getBurnedAmountByTx';
+import type { AvaxTx } from '../serializable/avax';
+import type { FeeState } from '../vms/pvm';
+import { validateEvmBurnedAmount } from './validateEvmBurnedAmount';
+import type { GetUpgradesInfoResponse } from '../info/model';
+import { isEtnaEnabled } from './isEtnaEnabled';
+import { validateAvaxBurnedAmountEtna } from './validateAvaxBurnedAmountEtna';
+import { validateAvaxBurnedAmountPreEtna } from './validateAvaxBurnedAmountPreEtna';
 import {
   isAvmBaseTx,
   isExportTx as isAvmExportTx,
   isImportTx as isAvmImportTx,
 } from '../serializable/avm';
-import { getBurnedAmountByTx } from './getBurnedAmountByTx';
-import type { AvaxTx } from '../serializable/avax';
-import { PrimaryNetworkID } from '../constants/networkIDs';
+import {
+  isAddDelegatorTx,
+  isAddValidatorTx,
+  isTransformSubnetTx,
+} from '../serializable/pvm';
 
 const _getBurnedAmount = (tx: Transaction, context: Context) => {
   const burnedAmounts = getBurnedAmountByTx(tx as AvaxTx | EVMTx);
   return burnedAmounts.get(context.avaxAssetID) ?? 0n;
 };
 
+// Transactions that are deprecated or not implemented for Etna
+// Todo: remove isAvmBaseTx, isAvmExportTx and isAvmImportTx when avm dynmamic fee is implemented
```

Comment:
Do you know whether x-chain will have dynamic pricing?

Reply:
my understanding is that it is in the road map but not gonna be implemented very soon.